cli/crates/common/src/environment.rs

```diff
@@ -330,6 +332,102 @@ fn find_grafbase_configuration(path: &Path, warnings: &mut Vec<Warning>) -> Opti
     }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub enum JavaScriptPackageManager {
+    Npm,
+    Pnpm,
+    Yarn,
+}
+
+impl Default for JavaScriptPackageManager {
+    fn default() -> Self {
+        Self::Npm
+    }
+}
+
+impl JavaScriptPackageManager {
+    #[must_use]
+    pub fn is_npm(self) -> bool {
+        matches!(self, Self::Npm)
+    }
+}
+
+impl fmt::Display for JavaScriptPackageManager {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Npm => f.write_str("npm"),
+            Self::Pnpm => f.write_str("pnpm"),
+            Self::Yarn => f.write_str("yarn"),
+        }
+    }
+}
+
+impl FromStr for JavaScriptPackageManager {
+    type Err = CommonError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "npm" => Ok(JavaScriptPackageManager::Npm),
+            "pnpm" => Ok(JavaScriptPackageManager::Pnpm),
+            "yarn" => Ok(JavaScriptPackageManager::Yarn),
+            _ => {
+                let inner = io::Error::new(
+                    io::ErrorKind::InvalidData,
+                    format!("Unsupported package manager in package.json: {s}"),
+                );
+
+                Err(CommonError::AccessPackageJson(inner))
+            }
+        }
+    }
+}
+
+pub fn install_dependencies(project_root: &Path) -> Result<(), CommonError> {
+    let command = match which::which("npm") {
```

Comment:
I'd check the behavior here if you already had it installed but with a different version than latest, it may overwrite

Reply:
We want to overwrite (discussed in Slack).